docs/core/v3.0.rst

```diff
@@ -936,22 +937,27 @@ containing the following names:
 ``metadata_encoding``
 ^^^^^^^^^^^^^^^^^^^^^
 
-    A string containing the URI pointing to a document describing the method
-    used for encoding group and array metadata documents.
-
-    For document using the default JSON encoding and format describe in this document
-    then the value must be ``"https://purl.org/zarr/spec/core/3.0``.
+    Specifies the encoding of group and array metadata. To use JSON encoding,
+    which is the only encoding allowed by this core specification, the
+    ``metadata_encoding`` value must be the following object::
 
-``metadata_key_suffix``
-^^^^^^^^^^^^^^^^^^^^^^^
+        {
+            "type": "json",
+            "metadata_key_suffix": ".json"
+        }
 
-    A string containing a suffix to add to the metadata keys when saving into
-    the store. By default ``".json"``.
+    The ``metadata_encoding`` value is an extension point and may be defined by
+    an extension. In this case the value must be an object containing the
+    required names ``extension``, ``type`` and ``metadata_key_suffix``.
+    ``extension`` must be a URI that identifies the extension and dereferences
+    to a human-readable representation of the specification. ``type`` is a
+    string defined by the extension. The ``metadata_key_suffix`` is a string
+    containing a suffix to add to the array and group metadata keys when saving
+    into the store.
```

Comment:
Question: is "metadata_key_suffix" allowed to be an empty string?

Reply:
I don't see any downsides, so why not? This would need to be defined in an extension though. See also #174